### PR TITLE
Use demo-snippet in demos.

### PR DESCRIPTION
**Pull request by bicknellr**
screen shot 2016-01-21 at 18 18 47
screen shot 2016-01-21 at 18 18 57
screen shot 2016-01-21 at 18 19 06
screen shot 2016-01-21 at 18 19 13
screen shot 2016-01-21 at 18 19 21
screen shot 2016-01-21 at 18 19 26
screen shot 2016-01-21 at 18 19 33

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -15,226 +15,213 @@
...
     <script src="../../webcomponentsjs/webcomponents-lite.js"></script>
 
-    <link rel="stylesheet" href="../../paper-styles/demo.css">
+    <link rel="import" href="../../iron-demo-helpers/demo-pages-shared-styles.html">
+    <link rel="import" href="../../iron-demo-helpers/demo-snippet.html">
     <link rel="import" href="../../polymer/polymer.html">
     <link rel="import" href="../iron-image.html">
 
+    <style is="custom-style" include="demo-pages-shared-styles"></style>
+    <style is="custom-style">
```
Comment (notwaldorf) [inline demo/index.html]: this style can just go in the previous one:
```
<style is="custom-style" include="demo-pages-shared-styles">
  demo-snippet.demo-hidden { ... }
</style>
```

File: demo/index.html [hunk 2]
```diff
@@ -15,226 +15,213 @@
...
+            (with <code>fade</code> attribute)
+          </div>
+        </template>
+      </demo-snippet>
+
+      <h3>(Used by all demos.)</h3>
+      <demo-snippet class="demo-hidden">
+        <template>
+          <style>
+            .example {
```
Comment (notwaldorf) [inline demo/index.html]: I think you can just move this into the main page style and not worry about it :)

File: bower.json [hunk 3]
```diff
@@ -25,7 +25,8 @@
     "paper-styles": "PolymerElements/paper-styles#^1.0.4",
     "test-fixture": "PolymerElements/test-fixture#^1.0.0",
     "web-component-tester": "^4.0.0",
-    "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0"
+    "webcomponentsjs": "webcomponents/webcomponentsjs#^0.7.0",
+    "iron-demo-helpers": "PolymerElements/iron-demo-helpers#^1.0.0"
```
Comment (notwaldorf) [inline bower.json]: maybe sort this alphabetically, since it looked sorted already?

**Comment by bicknellr** [comment]
@notwaldorf fixed stuff from your comments

**Comment by notwaldorf** [comment]
This is awesome! My only comment would be, if possible to use full, friendly sentences as the title headings. "Iron-image can be configured to take up the full height of a container" etc. 🍰

**Comment by bicknellr** [comment]
Updated the titles and slightly changed / expanded a few examples since I realized they were confusing after trying to actually explain them. Maybe worth another skim?

**Comment by notwaldorf** [comment]
I've started adding 
```
demo-snippet {
      --demo-snippet-code: {
        max-height: 300px;
      };
    }
``` 
to some demos since it was getting a bit annoying. Maybe that would work here too?

LGTM though! Super clear! :)

**Comment by bicknellr** [comment]
(from offline discussion) Waiting on the `max-height` thing because of this:
screen shot 2016-01-22 at 14 14 35

**Pull request merged**